### PR TITLE
Unicode string lit

### PR DESCRIPTION
**Pull request by sinhpham**
Fix https://github.com/nrc/rustfmt/issues/120

### CODE REVIEW

File: src/string.rs [hunk 1]
```diff
@@ -26,8 +31,11 @@ pub struct StringFormat<'a> {
 
 // TODO: simplify this!
 pub fn rewrite_string<'a>(s: &str, fmt: &StringFormat<'a>) -> String {
-    // FIXME I bet this stomps unicode escapes in the source string
     // TODO if lo.col > IDEAL - 10, start a new line (need cur indent for that)
+    let re = Regex::new(r"(\\[:space:]+)").unwrap();
+    let stripped_str = re.replace_all(s, "");
+
+    let graphemes = UnicodeSegmentation::graphemes(&*stripped_str, false).collect::<Vec<&str>>();
```
Comment (nrc) [inline src/string.rs]: do you need the `*` here?

File: src/string.rs [hunk 2]
```diff
@@ -26,8 +31,11 @@ pub struct StringFormat<'a> {
 
 // TODO: simplify this!
 pub fn rewrite_string<'a>(s: &str, fmt: &StringFormat<'a>) -> String {
-    // FIXME I bet this stomps unicode escapes in the source string
     // TODO if lo.col > IDEAL - 10, start a new line (need cur indent for that)
+    let re = Regex::new(r"(\\[:space:]+)").unwrap();
+    let stripped_str = re.replace_all(s, "");
```
Comment (nrc) [inline src/string.rs]: Could you comment these two lines please?

File: tests/source/string-lit.rs [hunk 3]
```diff
@@ -24,6 +24,10 @@ formatting"#;
 
     let xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx =
         funktion("yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy");
+        
+    let unicode = "a̐éö̲\r\n";
+    let unicode2 = "Löwe 老虎 Léopard";
+    let unicode3 = "中华Việt Nam";
```
Comment (nrc) [inline tests/source/string-lit.rs]: Could we have some tests with multi-line strings and unicode please?

**Comment by nrc** [comment]
Thanks for the PR! Looks like a good improvement, just needs some more comments and tests.

File: src/string.rs (hunk 1 above)
Comment (sinhpham) [inline src/string.rs]: Yes the UnicodeSegmentation::graphemes function is taking in a trait, not &str, so we need to deref a &String explicitly.

**Comment by nrc** [comment]
Thanks for the fixes!

**Pull request merged**